### ISSUE TITLE
device_names not complete due to parsing bug in FfmpegFormat.Reader._get_cam_inputname()

### ISSUE THREAD

**Issue by dennisvang** [imageio/imageio#283]
### Issue:
I am using imageio version 2.2.0 with ffmpeg v3.2.4 on a Windows 10 laptop with an onboard webcam and a video grabber attached through usb. A direct call to ffmpeg lists both video devices, but `imageio.get_reader('<video1>')` raises an IndexError.

### Origin:
I traced this back to the `device_list` in `FfmpegFormat.Reader._get_cam_inputname()`, which contains only one device instead of two.

### Cause:
This is caused by an assumption in the parsing algorithm.

Line 264 in ffmpeg.py relies on `"` being present at the start of a `line` containing a device name: 
    `if line.startswith('"'): ...`

However, ffmpeg v3.2.4 (on my system) also returns 'Alternative name: "......"' (see file [ffmpeg_device_list.txt](https://github.com/imageio/imageio/files/1321996/ffmpeg_device_list.txt)), which causes `in_video_devices` to be set to `False`. As it is never set `True` again, all subsequent video devices are skipped from the list.

### Proposed solution:
A patch that works for me is the following:
```
...
# Parse the result
device_names = []
in_video_devices = False
for line in infos.splitlines():
    if line.startswith('[dshow'):
        line = line.split(']', 1)[1].strip()
        if in_video_devices and line.startswith('"'):
            device_names.append(line[1:-1])
        elif 'video devices' in line:
            in_video_devices = True
        elif 'audio devices' in line:
            in_video_devices = False
...
```
Although I am not sure this covers all possible cases. This patch relies on another assumption, viz. that the `video devices` section is followed by an `audio devices` section.

**Comment by almarklein** [comment]
Thanks for looking into this! That code for parsing ffmpeg's output seems to be the most fragile bit of imageio :(  Could yo turn that patch into a PR? 

Can we detect the end of the video devices in a more reliable way? Can you please post a snippet of the ffmpeg output?

**Comment by dennisvang** [comment]
No problem. I'll try to create a pull request. 
I'm not sure if there is a more reliable way to do this... 
My ffmpeg output should already be available in the text file linked above.

**Comment by almarklein** [comment]
> I'll try to create a pull request.

Thanks. Let me know if you need help with that. Except for being easier for me :), it also gives you attribution for your contribution.

> My ffmpeg output should already be available in the text file linked above.

Sorry, I should've read better :)

**Comment by almarklein** [comment]
> I'm not sure if there is a more reliable way to do this...

Perhaps test whether the line had "devices" in it. Since its plural, it is very likely to be a "header".

**Comment by dennisvang** [comment]
Good idea. That should be a bit more robust than only looking for "audio devices".

Currently I have the parsing code in a separate method which takes the ffmpeg output (`infos`) as input argument and returns the `device_names` list. This should make it easier to test without actually having any webcams.  However, testing would require a sample like the one I posted above. I can hard-code that into the test, but I guess it would be neater to get it from an external file, e.g. from the imageio-binaries repo. 

What would you suggest?

**Comment by almarklein** [comment]
I think its a great idea to put these functions under test. I'd say just hardcode the (relevant parts of) the samples in the test itself.

I'd love similar tests for the other ffmpeg parsing, e.g. we've had trouble with collecting the fps as well.

**Comment by almarklein** [comment]
Fixed by  #284
